### ISSUE TITLE
Change the 30 Second Recording Limit

### ISSUE THREAD

**Issue by WesleyFister** [futo-org/voice-input#55]
Allow the user to change or disable the recording limit and by default if the user is holding the microphone button disable the limit.

**Comment by natarslan** [comment]
It would be very nice to be able to chose a recording limit or possibly a "stop recording" button.

**Comment by WesleyFister** [comment]
We already have a "stop recording" button. When recording the user can tap the microphone button to end the recording whenever they choose.

**Comment by natarslan** [comment]
Thats true. :) Im a new user so I missed that "stop button". Still, longer recording time would be great. Is it feasible to prolong the speaking time or is there a technical reason behind 30 sec?

**Comment by WesleyFister** [comment]
Whisper splits audio into 30 second chunks but that shouldn't be a limiting factor. Maybe, audio that is greater than 30 seconds experiences a massive slow down or something? I'm not entirely sure of the reason myself and would like to know this as well.

**Comment by natarslan** [comment]
Thank you for the quick reply. Yeah I can imagine that it has something to do with processing. Found [this](https://github.com/openai/whisper/discussions/1118#discussioncomment-5362741) post talking about the reason behind 30sec.

I also read [this](https://brunoscheufler.com/blog/2023-03-12-generating-subtitles-in-real-time-with-openai-whisper-and-pyaudio) project where one can record a longer audio and then split them into 30sec chunks and merge the transcribed result (if im understanding it correct). But perhaps its not a neat solution for futo.

**Comment by thibaultmol** [comment]
I really hope they can do an auto splitting thing so that we can indeed record longer than 30 seconds. 

I would love to use the better speech recognition of whisper to do my multi minute long brain dumping sessions using speech

**Comment by thibaultmol** [comment]
> Not that I like the idea of brain dumps but this would be a great use case. In the meantime, if you're on Linux you can use [SpeechNote](https://github.com/mkiol/dsnote) that would do the same thing but with bigger/better models.

Thanks for the recommendation! (I'm on linux) so I'll check it out.
Currently I'm just recording my voice and then putting it through Whishper https://github.com/pluja/whishper/

**Comment by WesleyFister** [comment]
Closing as feature was added in v1.3.6.
